src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1957,6 +1957,13 @@ namespace winrt::TerminalApp::implementation
             {
                 co_await winrt::resume_foreground(Dispatcher());
 
+                // We have to initialize the dialog here to be able to change the text of the text block within it
+                FindName(L"MultiLinePasteDialog").try_as<WUX::Controls::ContentDialog>();
+                ClipboardText().Text(text);
+
+                // The vertical offset on the scrollbar does not reset automatically, so reset it manually
+                ClipboardContentScrollViewer().ScrollToVerticalOffset(0);
+
                 ContentDialogResult warningResult;
```

Comment:
here's a thought- maybe after the dialog is dismissed, we should _clear_ `ClipboardText`. Otherwise, it will linger around in memory until the next multi-line paste. Thoughts?

Reply:
Makes sense to me, fixed!